### PR TITLE
added point in time table

### LINKED ISSUE #1998: Point in time - Stack Management Changes Phase 1
### Overview

For phase 1 of point in time UX implementation, we will be introducing a new section for 'point in time' under 'stack management' - reference : https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1612

### High level tasks:

- [ ] Point in time option in stack management side nav
- [ ] Table to track point in time searches
- [ ] Create PIT Side drawer
- [ ] Delete after expiry option in side drawer
- [ ] Rename PIT option
- [ ] Success / failures
- [ ] Discover page redirect
- [ ] Showing PIT details screen - general tab
- [ ] API integration -  List pits, delete pit and create PIT
- [ ] API integration -  Update PIT

### PR DESCRIPTION
**Pull request by ajygupta**
Signed-off-by: Ajay Gupta <ajyg@amazon.com>

### Description
Updated the Empty state with the point in time table. Added routing, breadcrumbs and data source selection support. 
There is integration with the backend APIs and saved objects to fetch PITs and populated the table which would be done in following PRs.
image

 
### Issues Resolved
#1998
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  useMount(() => {
+    setBreadcrumbs(getListBreadcrumbs());
+  });
+
+  // TODO: update this for cases when some data source name is default
+  const defaultDataSource: DataSourceItem = { id: '', title: 'Default', sort: '0' };
+
+  // TODO: use APIs to fetch PITs and update the table and message
+  const [pits] = useState([]);
+  const [message] = useState(<EmptyState />);
```
Comment (bharath-techie) [inline src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:48]: Why is this a component - not sure if we should use state to save components, can we use a boolean to either show zero state or table instead ?

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        className="pitTable"
+        horizontalPosition="center"
+        data-test-subj="pointInTimeTable"
+      >
+        <PageHeader />
+        <EuiSpacer size="m" />
+        <EuiPageContentBody>
+          <EuiInMemoryTable
+            items={pits}
+            itemId="name"
```
Comment (bharath-techie) [inline src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:168]: I see itemId="id" used in other pages - can we use the same for consistency?

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import { EmptyState } from './empty_state';
+import { PageHeader } from './page_header';
+
+export interface DataSourceItem {
+  id: string;
+  title: string;
+  sort: string;
+}
+
+const PITTable = ({ history }: RouteComponentProps) => {
```
Comment (bharath-techie) [inline src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:32]: Can we please add unit tests for this component ?

Comment (ajygupta) [reply src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:32]: Will add tests in the following PR when we add logic to fetch the PITs

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx (hunk 2 above)
Comment (ajygupta) [reply src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:168]: Will update this in following PR based on the property name we have in the PIT object.

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx (hunk 1 above)
Comment (ajygupta) [reply src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:48]: So we will be adding and using `setMessage` to update the message based on the response from the get PIT API. There can be other states like error fetching data.

**Comment by ajygupta** [comment]
@joshuarrrr @kavilla @ashwin-pc can you please review this change? Thanks

File: src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx (hunk 3 above)
Comment (ananzh) [reply src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx:32]: Yes, pls add more unit tests.

**Review by ananzh — COMMENTED** [review]
The "Point in Time Management" plugin adds a new functionality to OpenSearch Dashboards, which allows users to take a snapshot of their data at a specific point in time and then perform queries against that snapshot. The changes made in the pull request seem to be related to the plugin's UI, specifically in the src/plugins/point_in_time_management/public directory.

Have talked to @ajygupta and the project is targeting v2.8 release. Please feel free to add more pull request to my tracking issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3589 

Thank you!

**Comment by ananzh** [comment]
@ajygupta could you check my comments and make changes? thanks

**Comment by dblock** [comment]
Looks stalled, closing. @ajygupta please feel free to finish and reopen.

[Catch All Triage, attendees [1](https://github.com/dblock), [2](https://github.com/reta), [3](https://github.com/lukas-vlcek/), [4](https://github.com/rafaelma), [5](https://github.com/krisfreedain), [6](https://github.com/gargrajat), [7](https://github.com/andrross)]

**Pull request closed without merging**